contracts/TokenRules.sol

```diff
@@ -0,0 +1,264 @@
+pragma solidity ^0.4.24;
+pragma experimental ABIEncoderV2;
```

Comment:
As discussed, considering that experimental features will still be experimental by the time we intend to have these concepts on Mainnet, please make the necessary changes to not need to rely on those features.